src/test/java/com/google/sps/tests/UrlRequestTest.java

```diff
@@ -0,0 +1,139 @@
+package com.google.sps.data;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.Arrays;
+import java.util.List;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.HttpURLConnection;
+import java.net.MalformedURLException;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import org.powermock.api.mockito.PowerMockito;
+import org.hamcrest.MatcherAssert;
+import static org.hamcrest.collection.IsIterableContainingInAnyOrder.containsInAnyOrder;
+
+/** */
+@RunWith(PowerMockRunner.class) 
+@PrepareForTest({ URL.class, UrlRequest.class })
+public final class UrlRequestTest {
+
+  private final static String EMPTY_STRING = "";
+  private final static Map<String, String> EMPTY_PARAMS = new HashMap<>();
+  private final static String BASE_PATH = "http://www.testing.com";
+  private final static String URL_PATH = String.format("%s?%s", BASE_PATH, UrlRequest.getParamsString(EMPTY_PARAMS));
+
+  private static HttpURLConnection mockUrlConnection(String urlPath) throws Exception {
+    URL mockedUrl = PowerMockito.mock(URL.class);
+    HttpURLConnection mockedConnection = PowerMockito.mock(HttpURLConnection.class);
+
```

Comment:
[Nit] delete newline

Reply:
I believe this newline improves readability by dividing between mock creations and when events.